src/safeposix/syscalls/fs_calls.rs

```diff
@@ -1817,8 +1823,103 @@ impl Cage {
+                            let position = normalfile_filedesc_obj.position; // Get the current write position from the file descriptor object
+                            let filesize = normalfile_inode_obj.size; // Get the file size
+                            let blankbytecount = position as isize - filesize as isize; // Calculate the difference between the required and desired file position
+
+                            // Retrieve the file object from the file object table
+                            let mut fileobject = FILEOBJECTTABLE
+                                .get_mut(&normalfile_filedesc_obj.inode)
+                                .unwrap(); // Panic if the inode number does not exist
+
+                            // Pad the file with blank bytes if we are at a position past the end of the file
+                            if blankbytecount > 0 {
+                                if let Ok(byteswritten) = fileobject.zerofill_at(filesize, blankbytecount as usize) {
+                                    if byteswritten != blankbytecount as usize {
+                                        panic!("Write of blank bytes for writev failed!");
+                                    }
+                                } else {
+                                    panic!("Write of blank bytes for writev failed!");
+                                }
+                            }
+
+                            // Create IoSlice objects from the raw iovec pointer
+                            let iovs = iovec_to_ioslice(iovec, iovcnt);
+
+                            // Write to the file using the vectored IO method
+                            if let Ok(byteswritten) = fileobject.write_vectored_at(&iovs, position) {
+                                // Move position forward by the number of bytes we've written
+                                normalfile_filedesc_obj.position = position + byteswritten as usize;
+                                let newposition = normalfile_filedesc_obj.position;
+
+                                // Update file size if necessary
+                                if newposition > normalfile_inode_obj.size {
+                                    normalfile_inode_obj.size = newposition;
+                                    drop(inodeobj);
+                                    drop(fileobject);
+                                    log_metadata(&FS_METADATA, normalfile_filedesc_obj.inode);
+                                }
+
+                                byteswritten as i32
+                            } else {
+                                syscall_error(Errno::EIO, "writev", "Failed to write data to file")
+                            }
+                        }
+                        // Handle character device file
+                        Inode::CharDev(ref char_inode_obj) => {
+                            let iovecs = unsafe { std::slice::from_raw_parts(iovec, iovcnt as usize) };
```

Comment:
the buffer isnt actually ever written to here, i dont think we need to do all of this, can just send a blank buffer and leave a note to why

Reply:
updated